### DISCUSSION TITLE
Wagmi CLI returns an array instead of an object

### DISCUSSION THREAD

**Discussion by zkSoju** [wevm/wagmi#2320] [General]
Example:

```
  const { data: gateData } = useGatekeeperTokenToGates({
    address: GATEKEEPER_CONTRACT,
    args: [BigInt(BUNDLE_ID), BigInt(gateIndex)],
    enabled: !!address && gateIndex !== 5,
    watch: true,
  });
  ```
  
  Gate data used to provide the names for the variables making them more readable, but now it returns an array of types like `readonly [boolean, number, number, number, `0x${string}`, bigint]` is there a reason for this?

**Comment by jxom** [comment]
Yes! You can see the rationale here: https://github.com/wagmi-dev/viem/blob/main/.github/FAQ.md#why-is-a-contract-function-return-type-returning-an-array-instead-of-an-object

**Reply by zkSoju** [reply]
Got it thanks!

**Reply by zkSoju** [reply]
Perhaps we can have a flag to make these changes automatically?

**Reply by jxom** [reply]
Could make sense!

**Reply by zkSoju** [reply]
> Could make sense!

Another q!

If another framework (SnapshotJS) relies on the ethers `Signer` is there a way to make that work with WalletClient? Or would they have to upgrade to support WalletClient

**Reply by TarasHoneycomb** [reply]
Hey guys. Was this issue resolved or what? The link above is now broken
